src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -1,48 +1,88 @@
 import {observer, Observer} from "mobx-react";
-import {AlterationTypeConstants, ResultsViewPageStore} from "../ResultsViewPageStore";
+import {AlterationTypeConstants, ResultsViewPageStore, GeneticEntity, GeneticEntityType} from "../ResultsViewPageStore";
+import Select from "react-select";
 import internalClient from "../../../shared/api/cbioportalInternalClientInstance";
 import {CoExpression, CoExpressionFilter} from "../../../shared/api/generated/CBioPortalAPIInternal";
 import _ from "lodash";
 import {MSKTab, MSKTabs} from "../../../shared/components/MSKTabs/MSKTabs";
 import CoExpressionViz from "./CoExpressionViz";
 import LoadingIndicator from "shared/components/loadingIndicator/LoadingIndicator";
-import {CoExpressionWithQ, filterAndSortProfiles} from "./CoExpressionTabUtils";
+import {filterAndSortProfiles, getGenesetProfiles} from "./CoExpressionTabUtils";
 import MobxPromiseCache from "../../../shared/lib/MobxPromiseCache";
 import {ICoExpressionPlotProps} from "./CoExpressionPlot";
 import {bind} from "bind-decorator";
 import OqlStatusBanner from "../../../shared/components/oqlStatusBanner/OqlStatusBanner";
 import {getMobxPromiseGroupStatus} from "../../../shared/lib/getMobxPromiseGroupStatus";
-import MolecularProfileSelector from "../../../shared/components/MolecularProfileSelector";
+import {IDataQueryFilter} from "shared/lib/StoreUtils";
 
 export interface ICoExpressionTabProps {
     store:ResultsViewPageStore;
 }
 
-export class CoExpressionCache extends MobxPromiseCache<{entrezGeneId:number, molecularProfile:MolecularProfile, allData:boolean}, CoExpression[]> {};
+export class CoExpressionCache extends MobxPromiseCache<{profileX: MolecularProfile, 
+    profileY: MolecularProfile, geneticEntityId: string,
+    geneticEntityType: string, allData:boolean}, CoExpression[]> {}
 
 @observer
 export default class CoExpressionTab extends React.Component<ICoExpressionTabProps, {}> {
-    @observable _selectedMolecularProfile:MolecularProfile|undefined; // only undefined initially, until molecular profiles downloaded
-    @observable _selectedEntrezGeneId:number | undefined; // only undefined initially, until genes downloaded
+    @observable _selectedProfileX:string|undefined; // only undefined initially, until molecular profiles downloaded
+    @observable _selectedProfileY:string|undefined; // only undefined initially, until molecular profiles downloaded
+    @observable _selectedGeneticEntityId:string|undefined; // only undefined initially, until genes and gene sets downloaded
 
-    @computed get selectedMolecularProfile():MolecularProfile|undefined {
-        if (!this._selectedMolecularProfile && this.props.store.coexpressionTabMolecularProfiles.isComplete
-            && this.props.store.coexpressionTabMolecularProfiles.result.length > 0) {
-            return this.props.store.coexpressionTabMolecularProfiles.result[0];
+    @computed get selectedProfileX():string|undefined {
```

Comment:
all of these @computed wrappers around `_selectedSomething`s should be `remoteData`